packages/flutter/lib/src/rendering/object.dart

```diff
@@ -175,8 +175,6 @@ class PaintingContext extends ClipContext {
   /// into the layer subtree associated with this painting context. Otherwise,
   /// the child will be painted into the current PictureLayer for this context.
   void paintChild(RenderObject child, Offset offset) {
-    if (!kReleaseMode && debugProfilePaintsEnabled)
```

Comment:
Why are these getting moved into _paintWithContext? Shouldn't a child that is a repaintBoundary have the cost of being one associated with them?

Reply:
Both codepaths go through _paintWithContext when they do work. Mostly I was trying to bring this closer to the code that actually calls `paint()`, because I couldn't understand how it worked (at first I thought that there was no paint tracing and so I added it to _paintWithContext because that felt like the logical place for it). Then I discovered actually we already had code there, but I didn't understand that it was always called, so I removed it in favour of the new location. I could have just left it in the old location, I think the effect is more or less the same.

Let me know if you prefer the old location.

Reply:
The old location feels a little more appropriate to me, but I don't feel super strongly about this.

Reply:
i'll move it back

Reply:
oh, there is _one_ code path that doesn't go through paintChild, though, and that's the flushPaints loop.